simulator/mips/mips_instr.h

```diff
@@ -305,7 +309,29 @@ class FuncInstr
             if ( _is_jump_taken)
                 new_PC += sign_extend( v_imm) << 2;
         }
-
+        
+        void execute_clz()   
+        {
+            v_dst = 0;
+            for ( uint32_t i = 0x80000000; i > 0; i >>= 1)
```

Comment:
I don't like that solution as it introduces a branch here. Branches are harmful to performance as we now from our lectures. Integer-only solutions might be better.

In addition, there are x86 intrinsics for similar instruction, so it could be even converted into single instruction. But we don't go there as we want to keep our code portable.

Reply:
However, let's keep the loop for a while. We will optimize it later.